### PR TITLE
Misc doc and infra fix found in 2.55.0 release

### PR DESCRIPTION
**Pull request by Abacn**
* Release doc fixes

* Update workflow README for newly added workflows

* Do not trigger sickbay test in release verification

* Remove redundant shadowjar validation exclusions in harness build.gradle

* Use API token in deploy_rc_pypi action, in alignment with finalize release

**Please** add a meaningful description for your change here

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://github.com/apache/beam/blob/master/CONTRIBUTING.md#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI or the [workflows README](https://github.com/apache/beam/blob/master/.github/workflows/README.md) to see a list of phrases to trigger workflows.

### CODE REVIEW

File: .github/workflows/README.md [hunk 1]
```diff
@@ -215,8 +215,7 @@ PreCommit Jobs run in a schedule and also get triggered in a PR if relevant sour
...
 | [ PreCommit Java Examples Dataflow ](https://github.com/apache/beam/actions/workflows/beam_PreCommit_Java_Examples_Dataflow.yml) | N/A |`Run Java_Examples_Dataflow PreCommit`| [![.github/workflows/beam_PreCommit_Java_Examples_Dataflow.yml](https://github.com/apache/beam/actions/workflows/beam_PreCommit_Java_Examples_Dataflow.yml/badge.svg?event=schedule)](https://github.com/apache/beam/actions/workflows/beam_PreCommit_Java_Examples_Dataflow.yml?query=event%3Aschedule) |
-| [ PreCommit Java Examples Dataflow Java11 ](https://github.com/apache/beam/actions/workflows/beam_PreCommit_Java_Examples_Dataflow_Java11.yml) | N/A | `Run Java_Examples_Dataflow_Java11 PreCommit` | [![.github/workflows/beam_PreCommit_Java_Examples_Dataflow_Java11.yml](https://github.com/apache/beam/actions/workflows/beam_PreCommit_Java_Examples_Dataflow_Java11.yml/badge.svg?event=schedule)](https://github.com/apache/beam/actions/workflows/beam_PreCommit_Java_Examples_Dataflow_Java11.yml?query=event%3Aschedule) |
```
Comment (Abacn) [inline .github/workflows/README.md:218]: java11 and 17 precommits were moved to postcommit (line 292 of this file), and now only java(java8) and java21 run on precommit

File: .github/workflows/deploy_release_candidate_pypi.yaml [hunk 2]
```diff
@@ -13,11 +13,8 @@ on:
       RC:
         description: Integer RC version for the release (e.g. 3 for RC3)
         required: true
-      PYPI_USER:
-        description: PyPi username to perform the upload with
-        required: true
-      PYPI_PASSWORD:
-        description: PyPi password to perform the upload with
+      PYPI_API_TOKEN:
```
Comment (Abacn) [inline .github/workflows/deploy_release_candidate_pypi.yaml:16]: feeding username and password result in 403 ... legacy authentication rejected

The finalize release workflow already used API token. Align deploy rc workflow to it

File: sdks/python/apache_beam/examples/cookbook/bigtableio_it_test.py [hunk 3]
```diff
@@ -174,9 +174,7 @@ def tearDown(self):
     if self.instance.exists():
       self.instance.delete()
 
-  # TODO(https://github.com/apache/beam/issues/29076): Reenable this test
-  # once BigTable issues are fixed.
-  @pytest.mark.it_postcommit_sickbay
```
Comment (Abacn) [inline sdks/python/apache_beam/examples/cookbook/bigtableio_it_test.py:179]: This reverts #29197 as Python PostCommit sickbay has been passing this test, and the original todo issue is closed

**Review by damccorm — APPROVED** [review]
Thanks

**Pull request merged**